Update docs blazor version for .net core preview 3

The guide mentions to run this command:
```
dotnet new -i Microsoft.AspNetCore.Blazor.Templates::0.8.0-preview-19104-04
```
However, with the release of preview 3 today, that fails to build.

Update the docs to use the newest version of blazor templates

`0.9.0-preview3-19154-02`

---
#### Document details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: dbdcc295-2262-311c-99d4-7ece67c5f6ed
* Version Independent ID: 20152398-0971-0302-21f8-b17308f02183
* Content: [Get started with Blazor](https://docs.microsoft.com/en-gb/aspnet/core/client-side/spa/blazor/get-started?view=aspnetcore-3.0&tabs=visual-studio#feedback)
* Content Source: [aspnetcore/client-side/spa/blazor/get-started.md](https://github.com/aspnet/Docs/blob/master/aspnetcore/client-side/spa/blazor/get-started.md)
* Product: **aspnet-core**
* GitHub Login: @guardrex
* Microsoft Alias: **riande**

Running `dotnet new -i Microsoft.AspNetCore.Blazor.Templates::0.9.0-preview3-19154-02` allows me to create a working project using the CLI. The resulting project shows a number of warnings like this:

```
warning NU1603: Microsoft.AspNetCore.Blazor.Server 0.9.0-preview3-19121-06 depends on Microsoft.NETCore.Platforms (>= 3.0.0-preview.19109.6) but Microsoft.NETCore.Platforms 3.0.0-preview.19109.6 was not found. An approximate best match of Microsoft.NETCore.Platforms 3.0.0-preview3.19128.7 was resolved.
```

It seems to work correctly in some short tests.

Side note: the [ASP.NET Core Blazor Language Services](https://marketplace.visualstudio.com/items?itemName=aspnet.blazor) doesn't support preview 3 yet.

Second side note: using the working CLI project in VS doesn't work either. It fails with this error:

```
Error	NU1202	Package Microsoft.AspNetCore.Blazor.Cli 0.9.0-preview3-19121-06 is not compatible with netcoreapp2.2 (.NETCoreApp,Version=v2.2). Package Microsoft.AspNetCore.Blazor.Cli 0.9.0-preview3-19121-06 supports: netcoreapp3.0 (.NETCoreApp,Version=v3.0)	
```

--- a/aspnetcore/client-side/spa/blazor/get-started.md
+++ b/aspnetcore/client-side/spa/blazor/get-started.md
@@ -26,7 +26,7 @@ To create your first Blazor project in Visual Studio:
 1. Make the Blazor templates available for use with the .NET Core CLI by running the following command in a command shell:
 
    ```console
-   dotnet new -i Microsoft.AspNetCore.Blazor.Templates::0.8.0-preview-19104-04
+   dotnet new -i Microsoft.AspNetCore.Blazor.Templates::0.9.0-preview3-19154-02
    ```
 
 1. Select **File** > **New Project** > **Web** > **ASP.NET Core Web Application**.
@@ -91,7 +91,7 @@ Prerequisites:
 1. Add the Blazor templates by running the following command in a command shell:
 
    ```console
-   dotnet new -i Microsoft.AspNetCore.Blazor.Templates::0.8.0-preview-19104-04
+   dotnet new -i Microsoft.AspNetCore.Blazor.Templates::0.9.0-preview3-19154-02
    ```
 
 1. Create your first Blazor project in a command shell:

--- a/aspnetcore/razor-components/common/samples/3.x/BlazorSample/BlazorSample.csproj
+++ b/aspnetcore/razor-components/common/samples/3.x/BlazorSample/BlazorSample.csproj
@@ -8,10 +8,10 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.AspNetCore.Blazor" Version="0.8.0-preview-19104-04" />
-    <PackageReference Include="Microsoft.AspNetCore.Blazor.Build" Version="0.8.0-preview-19104-04" PrivateAssets="all" />
+    <PackageReference Include="Microsoft.AspNetCore.Blazor" Version="0.9.0-preview3-19154-02" />
+    <PackageReference Include="Microsoft.AspNetCore.Blazor.Build" Version="0.9.0-preview3-19154-02" PrivateAssets="all" />
 
-    <DotNetCliToolReference Include="Microsoft.AspNetCore.Blazor.Cli" Version="0.8.0-preview-19104-04" />
+    <DotNetCliToolReference Include="Microsoft.AspNetCore.Blazor.Cli" Version="0.9.0-preview3-19154-02" />
   </ItemGroup>
 
 </Project>

--- a/aspnetcore/razor-components/common/samples/3.x/BlazorSample/JsInteropClasses/ExampleJsInterop.cs
+++ b/aspnetcore/razor-components/common/samples/3.x/BlazorSample/JsInteropClasses/ExampleJsInterop.cs
@@ -10,26 +10,33 @@ namespace BlazorSample.JsInteropClasses
     #region snippet1
     public class ExampleJsInterop
     {
-        public static Task<string> Prompt(string text)
+        private readonly IJSRuntime _jsRuntime;
+
+        public ExampleJsInterop(IJSRuntime jsRuntime)
+        {
+            _jsRuntime = jsRuntime;
+        }
+
+        public Task<string> Prompt(string text)
         {
             // showPrompt is implemented in wwwroot/exampleJsInterop.js
-            return JSRuntime.Current.InvokeAsync<string>(
+            return _jsRuntime.InvokeAsync<string>(
                 "exampleJsFunctions.showPrompt",
                 text);
         }
 
-        public static Task<string> Display(string welcomeMessage)
+        public Task<string> Display(string welcomeMessage)
         {
             // displayWelcome is implemented in wwwroot/exampleJsInterop.js
-            return JSRuntime.Current.InvokeAsync<string>(
+            return _jsRuntime.InvokeAsync<string>(
                 "exampleJsFunctions.displayWelcome",
                 welcomeMessage);
         }
         
-        public static Task CallHelloHelperSayHello(string name)
+        public Task CallHelloHelperSayHello(string name)
         {
             // sayHello is implemented in wwwroot/exampleJsInterop.js
-            return JSRuntime.Current.InvokeAsync<object>(
+            return _jsRuntime.InvokeAsync<object>(
                 "exampleJsFunctions.sayHello",
                 new DotNetObjectRef(new HelloHelper(name)));
         }

--- a/aspnetcore/razor-components/common/samples/3.x/BlazorSample/Pages/JSInterop.cshtml
+++ b/aspnetcore/razor-components/common/samples/3.x/BlazorSample/Pages/JSInterop.cshtml
@@ -1,6 +1,6 @@
 @page "/JSInterop"
 @using BlazorSample.JsInteropClasses
-@using Microsoft.JSInterop;
+@inject IJSRuntime JSRuntime
 
 <h1>JavaScript Interop</h1>
 
@@ -15,8 +15,9 @@
 @functions {
     public async void TriggerJsPrompt()
     {
-        var name = await ExampleJsInterop.Prompt("What's your name?");
-        await ExampleJsInterop.Display($"Hello {name}! Welcome to Blazor!");
+        var exampleJsInterop = new ExampleJsInterop(JSRuntime);
+        var name = await exampleJsInterop.Prompt("What's your name?");
+        await exampleJsInterop.Display($"Hello {name}! Welcome to Blazor!");
     }
 }
 
@@ -64,6 +65,7 @@
 @functions {
     public async void TriggerNetInstanceMethod()
     {
-        await ExampleJsInterop.CallHelloHelperSayHello("Blazor");
+        var exampleJsInterop = new ExampleJsInterop(JSRuntime);
+        await exampleJsInterop.CallHelloHelperSayHello("Blazor");
     }
 }

--- a/aspnetcore/razor-components/common/samples/3.x/BlazorSample/wwwroot/exampleJsInterop.js
+++ b/aspnetcore/razor-components/common/samples/3.x/BlazorSample/wwwroot/exampleJsInterop.js
@@ -5,12 +5,12 @@
   displayWelcome: function (welcomeMessage) {
     document.getElementById('welcome').innerText = welcomeMessage;
   },
-    returnArrayAsyncJs: function () {
-      DotNet.invokeMethodAsync('BlazorSample', 'ReturnArrayAsync')
-        .then(data => {
-          data.push(4);
-            console.log(data);
-    })
+  returnArrayAsyncJs: function () {
+    DotNet.invokeMethodAsync('BlazorSample', 'ReturnArrayAsync')
+      .then(data => {
+        data.push(4);
+          console.log(data);
+    });
   },
   sayHello: function (dotnetHelper) {
     return dotnetHelper.invokeMethodAsync('SayHello')

--- a/aspnetcore/razor-components/javascript-interop.md
+++ b/aspnetcore/razor-components/javascript-interop.md
@@ -50,7 +50,6 @@ The following component:
 
 ```cshtml
 @page "/"
-@using Microsoft.JSInterop;
 @inject IJSRuntime JsRuntime;
 
 <h1>Call JavaScript Function Example</h1>
@@ -93,7 +92,53 @@ The following component:
 }
 ```
 
-For client-side Blazor apps, the `IJSRuntime` abstraction is accessible from `JSRuntime.Current`, which refers to the current user's request. Because there's only a single user of a client-side Blazor app, using `JSRuntime.Current` to invoke a JavaScript function works normally. Only use `JSRuntime.Current` in client-side Blazor apps.
+To use the `IJSRuntime` abstraction, adopt any of the following approaches:
+
+* Inject the `IJSRuntime` abstraction into the Razor file (*.cshtml*):
+
+  ```cshtml
+  @inject IJSRuntime JSRuntime
+
+  @functions {
+      public override void OnInit()
+      {
+          StocksService.OnStockTickerUpdated += stockUpdate =>
+          {
+              JSRuntime.InvokeAsync<object>(
+                  "handleTickerChanged", 
+                  stockUpdate.symbol, 
+                  stockUpdate.price);
+          };
+      }
+  }
+  ```
+* Inject the `IJSRuntime` abstraction into a class (*.cs*):
+
+  ```csharp
+  public class MyJsInterop
+  {
+      private readonly IJSRuntime _jsRuntime;
+
+      public MyJsInterop(IJSRuntime jsRuntime)
+      {
+          _jsRuntime = jsRuntime;
+      }
+
+      public Task<string> DoSomething(string data)
+      {
+          // The doSomething JavaScript method is implemented 
+          // in a JavaScript file, such as 'wwwroot/MyJsInterop.js'.
+          return _jsRuntime.InvokeAsync<string>(
+              "myJsFunctions.doSomething",
+              data);
+      }
+  }
+  ```
+* For dynamic content generation with `BuildRenderTree`, use the `[Inject]` attribute:
+
+  ```csharp
+  [Inject] IJSRuntime JSRuntime { get; set; }
+  ```
 
 In the client-side sample app that accompanies this topic, two JavaScript functions are available to the client-side app that interact with the DOM to receive user input and display a welcome message:
 
@@ -106,7 +151,7 @@ In the client-side sample app that accompanies this topic, two JavaScript functi
 
 Place the `<script>` tag that references the JavaScript file in the *wwwroot/index.html* file:
 
-[!code-html[](./common/samples/3.x/BlazorSample/wwwroot/index.html?highlight=16)]
+[!code-html[](./common/samples/3.x/BlazorSample/wwwroot/index.html?highlight=15)]
 
 Don't place a script tag in a component file because the script tag can't be updated dynamically.
 
@@ -116,7 +161,7 @@ The sample app uses a pair of C# methods, `Prompt` and `Display`, to invoke the 
 
 *JsInteropClasses/ExampleJsInterop.cs*:
 
-[!code-csharp[](./common/samples/3.x/BlazorSample/JsInteropClasses/ExampleJsInterop.cs?name=snippet1&highlight=6-8,14-16)]
+[!code-csharp[](./common/samples/3.x/BlazorSample/JsInteropClasses/ExampleJsInterop.cs?name=snippet1&highlight=13-15,21-23)]
 
 The `IJSRuntime` abstraction is asynchronous to allow for server-side scenarios. If the app runs client-side and you want to invoke a JavaScript function synchronously, downcast to `IJSInProcessRuntime` and call `Invoke<T>` instead. We recommend that most JavaScript interop libraries use the async APIs to ensure the libraries are available in all scenarios, client-side or server-side.
 
@@ -128,7 +173,7 @@ The sample app includes a component to demonstrate JS interop. The component:
 
 *Pages/JSInterop.cshtml*:
 
-[!code-cshtml[](./common/samples/3.x/BlazorSample/Pages/JsInterop.cshtml?start=1&end=21&highlight=2-3,9-11,13,16-20)]
+[!code-cshtml[](./common/samples/3.x/BlazorSample/Pages/JsInterop.cshtml?start=1&end=21)]
 
 1. When `TriggerJsPrompt` is executed by selecting the component's **Trigger JavaScript Prompt** button, the `ExampleJsInterop.Prompt` method in C# code is called.
 1. The `Prompt` method executes the JavaScript `showPrompt` function provided in the *wwwroot/exampleJsInterop.js* file.
@@ -180,9 +225,16 @@ namespace MyLib
 {
     public static class MyLibElementRefExtensions
     {
+        private readonly IJSRuntime _jsRuntime;
+
+        public MyJsInterop(IJSRuntime jsRuntime)
+        {
+            _jsRuntime = jsRuntime;
+        }
+
         public static Task Focus(this ElementRef elementRef)
         {
-            return JSRuntime.Current.InvokeAsync<object>("myLib.focusElement", elementRef);
+            return _jsRuntime.InvokeAsync<object>("myLib.focusElement", elementRef);
         }
     }
 }
@@ -219,7 +271,7 @@ The sample app includes a C# method to return an array of `int`s. The method is 
 
 *Pages/JsInterop.cshtml*:
 
-[!code-cshtml[](./common/samples/3.x/BlazorSample/Pages/JsInterop.cshtml?start=47&end=58&highlight=7-11)]
+[!code-cshtml[](./common/samples/3.x/BlazorSample/Pages/JsInterop.cshtml?start=48&end=59&highlight=7-11)]
 
 JavaScript served to the client invokes the C# .NET method.
 
@@ -246,13 +298,13 @@ When the **Trigger .NET instance method HelloHelper.SayHello** button is selecte
 
 *Pages/JsInterop.cshtml*:
 
-[!code-cshtml[](./common/samples/3.x/BlazorSample/Pages/JsInterop.cshtml?start=60&end=69&highlight=8)]
+[!code-cshtml[](./common/samples/3.x/BlazorSample/Pages/JsInterop.cshtml?start=61&end=71&highlight=8-9)]
 
 `CallHelloHelperSayHello` invokes the JavaScript function `sayHello` with a new instance of `HelloHelper`.
 
 *JsInteropClasses/ExampleJsInterop.cs*:
 
-[!code-csharp[](./common/samples/3.x/BlazorSample/JsInteropClasses/ExampleJsInterop.cs?name=snippet1&highlight=19-25)]
+[!code-csharp[](./common/samples/3.x/BlazorSample/JsInteropClasses/ExampleJsInterop.cs?name=snippet1&highlight=26-32)]
 
 *wwwroot/exampleJsInterop.js*:
 

--- a/aspnetcore/razor-components/javascript-interop.md
+++ b/aspnetcore/razor-components/javascript-interop.md
@@ -18,7 +18,7 @@ A Razor Components app can invoke JavaScript functions from .NET and .NET method
 
 There are times when .NET code is required to call a JavaScript function. For example, a JavaScript call can expose browser capabilities or functionality from a JavaScript library to the app.
 
-To call into JavaScript from .NET, use the `IJSRuntime` abstraction. The `InvokeAsync<T>` method on `IJSRuntime` takes an identifier for the JavaScript function you wish to invoke along with any number of JSON-serializable arguments. The function identifier is relative to the global scope (`window`). If you wish to call `window.someScope.someFunction`, the identifier is `someScope.someFunction`. There's no need to register the function before it's called. The return type `T` must also be JSON serializable.
+To call into JavaScript from .NET, use the `IJSRuntime` abstraction. The `InvokeAsync<T>` method takes an identifier for the JavaScript function that you wish to invoke along with any number of JSON-serializable arguments. The function identifier is relative to the global scope (`window`). If you wish to call `window.someScope.someFunction`, the identifier is `someScope.someFunction`. There's no need to register the function before it's called. The return type `T` must also be JSON serializable.
 
 For server-side ASP.NET Core Razor Components apps:
 
@@ -41,7 +41,11 @@ Inside the `<head>` element of *wwwroot/index.html*, provide a function that use
 </script>
 ```
 
-JavaScript code, such as the code shown in the preceding example, can also be loaded by a JavaScript file with a reference to the script file in the *wwwroot/index.html* file.
+JavaScript code, such as the code shown in the preceding example, can also be loaded from a JavaScript file (*.js*) with a reference to the script file in the *wwwroot/index.html* file:
+
+```html
+<script src="exampleJsInterop.js"></script>
+```
 
 The following component:
 
@@ -153,9 +157,9 @@ Place the `<script>` tag that references the JavaScript file in the *wwwroot/ind
 
 [!code-html[](./common/samples/3.x/BlazorSample/wwwroot/index.html?highlight=15)]
 
-Don't place a script tag in a component file because the script tag can't be updated dynamically.
+Don't place a `<script>` tag in a component file because the `<script>` tag can't be updated dynamically.
 
-.NET methods interop with the JavaScript functions by calling `InvokeAsync<T>` method on `IJSRuntime`.
+.NET methods interop with the JavaScript functions in the *exampleJsInterop.js* file by calling `IJSRuntime.InvokeAsync<T>`.
 
 The sample app uses a pair of C# methods, `Prompt` and `Display`, to invoke the `showPrompt` and `displayWelcome` JavaScript functions:
 
@@ -163,13 +167,13 @@ The sample app uses a pair of C# methods, `Prompt` and `Display`, to invoke the 
 
 [!code-csharp[](./common/samples/3.x/BlazorSample/JsInteropClasses/ExampleJsInterop.cs?name=snippet1&highlight=13-15,21-23)]
 
-The `IJSRuntime` abstraction is asynchronous to allow for server-side scenarios. If the app runs client-side and you want to invoke a JavaScript function synchronously, downcast to `IJSInProcessRuntime` and call `Invoke<T>` instead. We recommend that most JavaScript interop libraries use the async APIs to ensure the libraries are available in all scenarios, client-side or server-side.
+The `IJSRuntime` abstraction is asynchronous to allow for server-side scenarios. If the app runs client-side and you want to invoke a JavaScript function synchronously, downcast to `IJSInProcessRuntime` and call `Invoke<T>` instead. We recommend that most JavaScript interop libraries use the async APIs to ensure that the libraries are available in all scenarios, client-side or server-side.
 
-The sample app includes a component to demonstrate JS interop. The component:
+The sample app includes a component to demonstrate JavaScript interop. The component:
 
-* Receives user input via a JS prompt.
+* Receives user input via a JavaScript prompt.
 * Returns the text to the component for processing.
-* Calls a second JS function that interacts with the DOM to display a welcome message.
+* Calls a second JavaScript function that interacts with the DOM to display a welcome message.
 
 *Pages/JSInterop.cshtml*:
 
@@ -189,7 +193,7 @@ You can capture references to HTML elements in a component by adding a `ref` att
 
 The following example shows capturing a reference to the username input element:
 
-```csharp
+```cshtml
 <input ref="username" ... />
 
 @functions {
@@ -200,7 +204,7 @@ The following example shows capturing a reference to the username input element:
 > [!NOTE]
 > Do **not** use captured element references as a way of populating the DOM. Doing so may interfere with the declarative rendering model.
 
-As far as .NET code is concerned, an `ElementRef` is an opaque handle. The *only* thing you can do with it is pass it through to JavaScript code via JavaScript interop. When you do so, the JavaScript-side code receives an `HTMLElement` instance, which it can use with normal DOM APIs.
+As far as .NET code is concerned, an `ElementRef` is an opaque handle. The *only* thing you can do with `ElementRef` is pass it through to JavaScript code via JavaScript interop. When you do so, the JavaScript-side code receives an `HTMLElement` instance, which it can use with normal DOM APIs.
 
 For example, the following code defines a .NET extension method that enables setting the focus on an element:
 
@@ -234,13 +238,14 @@ namespace MyLib
 
         public static Task Focus(this ElementRef elementRef)
         {
-            return _jsRuntime.InvokeAsync<object>("myLib.focusElement", elementRef);
+            return _jsRuntime.InvokeAsync<object>(
+                "myLib.focusElement", elementRef);
         }
     }
 }
 ```
 
-Now you can focus inputs in any of your components:
+Use `MyLib` and call `Focus` on an `ElementRef` to focus inputs in any component:
 
 ```cshtml
 @using MyLib
@@ -265,7 +270,7 @@ Now you can focus inputs in any of your components:
 
 ### Static .NET method call
 
-To invoke a static .NET method from JavaScript, use the `DotNet.invokeMethod` or `DotNet.invokeMethodAsync` functions. Pass in the identifier of the static method you wish to call, the name of the assembly containing the function, and any arguments. Again, the async version is preferred to support server-side scenarios. To be invokable from JavaScript, the .NET method must be public, static, and decorated with `[JSInvokable]`. By default, the method identifier is the method name, but you can specify a different identifier using the `JSInvokableAttribute` constructor. Calling open generic methods isn't currently supported.
+To invoke a static .NET method from JavaScript, use the `DotNet.invokeMethod` or `DotNet.invokeMethodAsync` functions. Pass in the identifier of the static method you wish to call, the name of the assembly containing the function, and any arguments. The asynchronous version is preferred to support server-side scenarios. To be invokable from JavaScript, the .NET method must be public, static, and decorated with `[JSInvokable]`. By default, the method identifier is the method name, but you can specify a different identifier using the `JSInvokableAttribute` constructor. Calling open generic methods isn't currently supported.
 
 The sample app includes a C# method to return an array of `int`s. The method is decorated with the `JSInvokable` attribute.
 
@@ -279,7 +284,9 @@ JavaScript served to the client invokes the C# .NET method.
 
 [!code-javascript[](./common/samples/3.x/BlazorSample/wwwroot/exampleJsInterop.js?highlight=8-12)]
 
-When the **Trigger .NET static method ReturnArrayAsync** button is selected, examine the console output in the browser's web developer tools:
+When the **Trigger .NET static method ReturnArrayAsync** button is selected, examine the console output in the browser's web developer tools.
+
+The console output is:
 
 ```console
 Array(4) [ 1, 2, 3, 4 ]
@@ -324,8 +331,10 @@ Hello, Blazor!
 
 ## Share interop code in a Razor Component class library
 
-JavaScript interop code can be included in a Razor Component class library (`dotnet new blazorlib`), which allows you to share the code in a NuGet package.
+JavaScript interop code can be included in a Razor Component class library (`dotnet new razorclasslib`), which allows you to share the code in a NuGet package.
 
-The Razor Component class library handles embedding JavaScript resources in the built assembly. The JavaScript files are placed in the *wwwroot* folder, and the tooling takes care of embedding the resources when the library is built.
+The Razor Component class library handles embedding JavaScript resources in the built assembly. The JavaScript files are placed in the *wwwroot* folder. The tooling takes care of embedding the resources when the library is built.
 
-The built NuGet package is referenced in the project file of the app just as any normal NuGet package is referenced. After the app has been restored, app code can call into JavaScript as if it were C#.
+The built NuGet package is referenced in the project file of the app just as any normal NuGet package is referenced. After the app is restored, app code can call into JavaScript as if it were C#.
+
+For more information, see <xref:razor-components/class-libraries>.

